### PR TITLE
refactor: pass arguments through to native methods

### PR DESCRIPTION
**Pull request by ekashida**
Pass arguments down intact where possible to facilitate native validation.

### CODE REVIEW

File: packages/@lwc/synthetic-shadow/src/polyfills/event-target/polyfill.ts [hunk 1]
```diff
@@ -17,55 +17,62 @@ import {
...
+    _listener: EventListenerOrEventListenerObject,
+    _optionsOrCapture?: boolean | AddEventListenerOptions
 ) {
-    if (arguments.length < 2) {
-        // Delegate to the browser for throwing these errors.
+    if (isHostElement(this)) {
+        // Typescript does not like it when you treat the `arguments` object as an array
         // @ts-ignore type-mismatch
-        return nativeAddEventListener.apply(this, arguments);
+        return addCustomElementEventListener.apply(this, arguments);
```
Comment (pmdartus) [inline packages/@lwc/synthetic-shadow/src/polyfills/event-target/polyfill.ts:29]: I don't think is right here: `addCustomElementEventListener` expects the element as the first argument and not as this value. It expects the following arguments today.

```suggestion
        return addCustomElementEventListener(this, ...arguments);
```

The `addCustomElementEventListener` does also some validation internally. Those validations should be removed.

File: packages/@lwc/synthetic-shadow/src/polyfills/event-target/polyfill.ts [hunk 2]
```diff
@@ -17,55 +17,62 @@ import {
...
+    _listener: EventListenerOrEventListenerObject,
+    _optionsOrCapture?: boolean | EventListenerOptions
 ) {
-    if (arguments.length < 2) {
-        // Delegate to the browser for throwing these errors.
+    if (isHostElement(this)) {
+        // Typescript does not like it when you treat the `arguments` object as an array
         // @ts-ignore type-mismatch
-        return nativeAddEventListener.apply(this, arguments);
+        return removeCustomElementEventListener.apply(this, arguments);
```
Comment (pmdartus) [inline packages/@lwc/synthetic-shadow/src/polyfills/event-target/polyfill.ts:52]: Same issue with `removeCustomElementEventListener`

File: packages/@lwc/synthetic-shadow/src/shared/event-target.ts [hunk 3]
```diff
@@ -47,9 +57,8 @@ export function getEventListenerWrapper(
             }
 
             return isFunction(fnOrObj)
-                ? (fnOrObj as EventListener).call(this, event)
-                : (fnOrObj as EventListenerObject).handleEvent &&
-                      (fnOrObj as EventListenerObject).handleEvent(event);
+                ? fnOrObj.call(this, event)
+                : fnOrObj.handleEvent && fnOrObj.handleEvent(event);
```
Comment (pmdartus) [inline packages/@lwc/synthetic-shadow/src/shared/event-target.ts:61]:
```suggestion
                : fnOrObj.handleEvent?.(event);
```

Comment (ekashida) [reply packages/@lwc/synthetic-shadow/src/shared/event-target.ts:61]: Didn't we revert ES2020 support?

Comment (pmdartus) [reply packages/@lwc/synthetic-shadow/src/shared/event-target.ts:61]: This is done by the TypeScript compiler so it should be fine.

**Pull request merged**